CharlieBackend.Api.UnitTes/StudentGroupServiceTest.cs

```diff
@@ -0,0 +1,241 @@
+                StudentIds = new List<long>() ,
+                MentorIds = new List<long>() 
+
+            };
+
+            var existingNameStudentGroupDto = new UpdateStudentGroupDto()
+            {
+                Name = "Test_name",
+                CourseId = 4,
+                StartDate = DateTime.Now,
+                FinishDate = DateTime.Now.AddMonths(3),
+                StudentIds = new List<long>(),
+                MentorIds = new List<long>()
+            };
+
+            var existingStudentGroup = new StudentGroup()
+            {
+                Id = 10,
+                Name = "Test_name",
+                CourseId = 3,
+                StartDate = DateTime.Now.AddMonths(3),
+                FinishDate = DateTime.Now.AddMonths(6),
+                StudentsOfStudentGroups = new List<StudentOfStudentGroup>(),
+                MentorsOfStudentGroups = new List<MentorOfStudentGroup>()
+            };
+
+            var existingCourse = new Course()
+            {
+                Id = 2,
+                Name = "AAA"
+            };
+
+            var studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
+            var studentsRepositoryMock = new Mock<IStudentRepository>();
+            var mentorsRepositoryMock = new Mock<IMentorRepository>();
+            var courseRepositoryMock = new Mock<ICourseRepository>();
+
+            studentGroupRepositoryMock.Setup(x => x.GetByIdAsync(existingStudentGroup.Id))
+                .ReturnsAsync(existingStudentGroup);
+
+            studentGroupRepositoryMock.Setup(x => x.GetByIdAsync(notExistingGroupId))
+                .ReturnsAsync((StudentGroup)null);
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(updateStudentGroupDto.Name))
+                .ReturnsAsync(false);
+
+            studentGroupRepositoryMock.Setup(x => x.IsGroupNameExistAsync(existingNameStudentGroupDto.Name))
+               .ReturnsAsync(true);
+
+            courseRepositoryMock.Setup(x => x.GetByIdAsync(updateStudentGroupDto.CourseId))
+              .ReturnsAsync(existingCourse);
+
+            studentGroupRepositoryMock.Setup(x => x.UpdateManyToMany(It.IsAny<IList<StudentOfStudentGroup>>(),
+                                                                     It.IsAny<IList<StudentOfStudentGroup>>()));
+
+            mentorsRepositoryMock.Setup(x => x.UpdateMentorGroups(It.IsAny<IList<MentorOfStudentGroup>>(),
+                                                                  It.IsAny<IList<MentorOfStudentGroup>>()));
+
+            _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.StudentRepository).Returns(studentsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorsRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.CourseRepository).Returns(courseRepositoryMock.Object);
```

Comment:
too much symbols in the lines

Reply:
It is divided into same logical block, spaces can harm the logical structure of the code